src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
+		List<GKInstance> currentRLEAuthoredInstances = currentRLE.getAttributeValuesList(ReactomeJavaConstants.authored);
+		List<GKInstance> currentRLERevisedInstances = currentRLE.getAttributeValuesList(ReactomeJavaConstants.revised);
+		List<GKInstance> previousRLEReviewedInstances = previousRLE.getAttributeValuesList(ReactomeJavaConstants.reviewed);
+		List<GKInstance> previousRLEAuthoredInstances = previousRLE.getAttributeValuesList(ReactomeJavaConstants.authored);
+		List<GKInstance> previousRLERevisedInstances = previousRLE.getAttributeValuesList(ReactomeJavaConstants.revised);
+
+		// Check if any of those attributes have a new instance edit
+		if (hasNewInstanceEdit(currentRLEReviewedInstances, previousRLEReviewedInstances)
+			|| hasNewInstanceEdit(currentRLEAuthoredInstances, previousRLEAuthoredInstances)
+			|| hasNewInstanceEdit(currentRLERevisedInstances, previousRLERevisedInstances))
+		{
 
-					boolean hasRLEInstanceEdit = false;
+			// Filter out any instance edits (IE) created by someone from Reactome.
+			// This is because Pathways are only officially considered updated once they are
+			// reviewed by an external person.
+			List<GKInstance> newInstanceEdits = getNewNonReactomeInstanceEdits(currentRLE, currentRLEReviewedInstances, currentRLEAuthoredInstances, currentRLERevisedInstances, previousRLEReviewedInstances, previousRLEAuthoredInstances, currentRLERevisedInstances);
 
-					// Iterate through all new instance edits, trying to find a Pathway containing
-					// them.
-					for (GKInstance rleIE : newRLEInstanceEdits)
+			if (!newInstanceEdits.isEmpty())
+			{
+				// Now it will check the immediate parent Pathway of the RlE to see if the new
+				// IE exists there.
+				// If it is not found, it will continue to check up the Pathway hierarchy to see
+				// if a new IE exists anywhere.
+
+				// Note: Expand so that instances with multiple parents are handled.
+				GKInstance parentPathway = (GKInstance) currentRLE.getReferers(ReactomeJavaConstants.hasEvent).iterator().next();
+				List<GKInstance> parentPathwayReviewedAuthoredRevisedInstances = parentPathway.getAttributeValuesList(ReactomeJavaConstants.reviewed);
+				parentPathwayReviewedAuthoredRevisedInstances.addAll(parentPathway.getAttributeValuesList(ReactomeJavaConstants.authored));
+				parentPathwayReviewedAuthoredRevisedInstances.addAll(parentPathway.getAttributeValuesList(ReactomeJavaConstants.revised));
+
+				for (GKInstance newIE : newInstanceEdits)
+				{
+					for (GKInstance parentPathwayIE : parentPathwayReviewedAuthoredRevisedInstances)
 					{
-						for (GKInstance pathwayIE : parentPathwayIEs)
-						{
-							if (rleIE.getDBID().equals(pathwayIE.getDBID()))
-							{
-								// The parent Pathway contained the IE.
-								hasRLEInstanceEdit = true;
-							}
-						}
-						if (hasRLEInstanceEdit)
+
+						// At time of writing, no 'existing' RlEs have a new IE in the immediate parent
```

Comment:
Could a date be included after "At time of writing"?

Reply:
No, because I didn't write that so I don't know what the actual time of that writing was.